src/components/portal/portal.tsx

```diff
@@ -141,6 +141,9 @@ export class Portal {
         this.container = document.createElement('div');
         this.container.setAttribute('id', this.containerId);
         this.container.setAttribute('class', 'limel-portal--container');
+        Object.assign(this.container, {
+            portalSource: this.host,
+        });
```

Comment:
Why do we need to use `Object.assign` here?

Reply:
It's kind of a cheat to trick the compiler :smile: 

The following will give a compiler error since `portalSource` is not part of `HTMLElement`
```ts
this.container.portalSource = this.host;
```

The following will give a lint error since we should use the syntax from the first example :upside_down_face: 
```ts
this.container['portalSource'] = this.host;
```

Reply:
We could introduce another internal web-component, e.g. `limel-portal-container` instead of using a regular `div` for the container which would make this a bit more clear, but that's for a future project if we want to do that :slightly_smiling_face:

Reply:
Thanks! I guessed it boiled down to that but wanted to ensure that I didn't misunderstand some parts 

Reply:
Another component might also separate the logic a bit which could be nice :thinking: 
The portal would be responsible for moving the elements in the DOM and the container would be responsible for positioning them